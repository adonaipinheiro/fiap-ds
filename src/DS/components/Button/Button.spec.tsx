import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { Button } from ".";
import { FaBeer } from "react-icons/fa";

describe("Button Component", () => {
  it("should render with children text", () => {
    render(<Button>Click me</Button>);

    expect(screen.getByText("Click me")).toBeTruthy();
  });

  it("should render with left icon", () => {
    render(<Button leftIcon={<FaBeer />}>Click me</Button>);

    expect(screen.getByTestId("icon-container")).toBeTruthy();
    expect(screen.getByText("Click me")).toBeTruthy();
  });

  it("should be disabled when passed the disabled prop", () => {
    render(<Button disabled>Click me</Button>);

    const button = screen.getByText("Click me");
    expect(button).toBeTruthy();
  });

  it("should take a snapshot", () => {
    const { asFragment } = render(<Button>Click me</Button>);

    expect(asFragment()).toMatchSnapshot();
  });
});
