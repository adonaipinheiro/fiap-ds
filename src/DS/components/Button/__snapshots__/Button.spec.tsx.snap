// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-blHHSb gjEoJm"
  >
    Click me
  </button>
</DocumentFragment>
`;
